libbeat/publisher/event.go

```diff
@@ -41,11 +44,45 @@ type Batch interface {
 type Event struct {
 	Content beat.Event
 	Flags   EventFlags
+	Cache   *EventCache
```

Comment:
having a pointer here will lead to a many calls with a nil pointer. Did you change it into a pointer because of mutex not being copyable?

Reply:
Yes, that was the reason I changed it. Perhaps there's a better solution?

Reply:
The easiest would be to remove the mutex and assume that Cache can only be accessed by one output at a time. This way we only need to allocate the cache if required.

The publisher.Event uses copy semantics. This means that the lazy init if the Cache is initialized lazily might not work properly. Sure it creates a map, but only in the copy of the publisher.Event. If output code passes the copied value to another function and then initialized the cache, then the cache will not be shared at all.

In order to guarantee sharing always (make it a little more fool-proof) we need to allocated/initialize somewhere.  Either in `newBatch` in the publisher pipeline or in the queue lib. As batches are ACKed and must not be reused, we could but cache instances into a `sync.Pool`.